src/main/java/ca/corefacility/bioinformatics/irida/repositories/filesystem/IridaFileStorageAzureUtilityImpl.java

```diff
@@ -0,0 +1,281 @@
+import java.io.InputStream;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.Optional;
+import java.util.zip.GZIPInputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import ca.corefacility.bioinformatics.irida.exceptions.StorageException;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequenceFile;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
+import ca.corefacility.bioinformatics.irida.util.FileUtils;
+
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+import com.azure.storage.blob.models.BlobStorageException;
+
+/**
+ * Component implementation of file utitlities for azure storage
+ */
+@Component
+public class IridaFileStorageAzureUtilityImpl implements IridaFileStorageUtility {
+	private static final Logger logger = LoggerFactory.getLogger(IridaFileStorageAzureUtilityImpl.class);
+
+	private BlobServiceClient blobServiceClient;
+	private BlobContainerClient containerClient ;
+
+	@Autowired
+	public IridaFileStorageAzureUtilityImpl(String connectionStr, String containerName){
+		this.blobServiceClient = new BlobServiceClientBuilder().connectionString(connectionStr)
+				.buildClient();
+		this.containerClient = blobServiceClient.getBlobContainerClient(containerName);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public File getTemporaryFile(Path file) {
+		File fileToProcess = null;
+
+		// We set the blobClient "path" to which file we want to get
+		BlobClient blobClient = containerClient.getBlobClient(getAzureFileAbsolutePath(file));
+
+		try {
+			logger.trace("Getting file from azure [" + file.toString() + "]");
+			Path tempDirectory = Files.createTempDirectory(null);
+			Path tempFile = tempDirectory.resolve(file.getFileName().toString());
+			InputStream initialStream = blobClient.openInputStream();
+			org.apache.commons.io.FileUtils.copyInputStreamToFile(initialStream, tempFile.toFile());
+			initialStream.close();
+			fileToProcess = tempFile.toFile();
+		} catch (BlobStorageException e) {
+			logger.error("Couldn't find file on azure [" + e + "]");
+			throw new StorageException("Unable to locate file on azure", e);
+		} catch (IOException e) {
+			logger.error(e.getMessage());
+			throw new StorageException(e.getMessage());
+		}
+
+		return fileToProcess;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void cleanupLocalFiles(Path path) {
+		logger.trace("Cleaning up temporary file downloaded from azure [" + path.toString() + "]");
+
+		Path origPath = path;
+		if(Files.isRegularFile(path)) {
+			origPath = path;
+			path = path.getParent();
```

Comment:
I don't think we should walk up to delete a parent directory.  While this shouldn't be an issue with our temp directories, it has the possibility of being accidentally misused and deleting a required directory.  if we're deleting a directory, I think it should be explicitly called to be deleted.

Reply:
Updated in 05feefc11ed7c0e21deb78ad450036145ace7e5f